scripts/install-mender.sh

```diff
@@ -152,21 +195,82 @@ add_repo() {
     echo "$repo" > /etc/apt/sources.list.d/mender.list
 }
 
-do_install() {
+do_install_open() {
+    # Filter out commercial components
+    local selected_components_open=""
+    for c in $SELECTED_COMPONENTS
+    do
+        if ! echo "$COMMERCIAL_COMPONENTS" | egrep -q "(^| )$c( |\$)"; then
+            selected_components_open="$selected_components_open $c"
+        fi
+    done
+
+    # Return if no open source components selected
+    if [ -z "$selected_components_open" ]; then
+        return
+    fi
+
+    echo "  Installing open source components from APT repository"
+
     apt-get update
     apt-get install -y \
        -o Dpkg::Options::="--force-confdef" \
        -o Dpkg::Options::="--force-confold" \
-       $SELECTED_COMPONENTS
+       $selected_components_open
+
+    echo "  Success! Please run \`mender setup\` to configure the client."
+}
+
+do_install_commercial() {
+    # Filter commercial components
+    local selected_components_commercial=""
+    for c in $SELECTED_COMPONENTS
+    do
+        if echo "$COMMERCIAL_COMPONENTS" | egrep -q "(^| )$c( |\$)"; then
+            selected_components_commercial="$selected_components_commercial $c"
+        fi
+    done
+
+    # Return if no commercial components selected
+    if [ -z "$selected_components_commercial" ]; then
+        return
+    fi
+
+    echo "  Installing commercial components from $MENDER_COMMERCIAL_DOWNLOAD_URL"
+
+    # Translate Debian "channel" into Mender version
+    if [ "$CHANNEL" = "experimental" ]; then
+        version="master"
+    else
+        version="latest"
+    fi
+
+    # Download deb packages
+    for c in $selected_components_commercial
+    do
+        url="$MENDER_COMMERCIAL_DOWNLOAD_URL$(printf ${COMMERCIAL_COMP_TO_URL_PATH_F[$c]} $version $version)"
+        curl -fLsS -H "Authorization: Bearer $JWT_TOKEN" -O $url
```

Comment:
suggested change $url -> "$url"

Reply:
shouldnt we check is the `basename $url` exists as a file, i.e.: the donwload was successful, andif not print the relevant error?

Reply:
:+1: